File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema02Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import org.junit.jupiter.api.Test

/**
 * @author benjamin.krenn@leftshift.one - 10/25/19.
 * @since 0.1.0
 */
class Schema02Test : AbstractDynabuffersTest() {

    @Test
    fun testParse() {
        val engine = Dynabuffers.parse(Schema02Test::class.java.getResourceAsStream("/schema02.dbs"))

        val input = mapOf(
                "message" to mapOf("wasteType" to "Schuhe unbeschädigt", "point" to mapOf("lat" to 47.070890f, "lng" to 15.439279f)))

        val output = mapOf(
                "message" to mapOf("wasteType" to "Schuhe unbeschädigt", "point" to mapOf("lat" to 47.070890f, "lng" to 15.439279f), "_type" to 0))

        assertResultMap(engine, input, output)
    }

}

File: dynabuffers-python/tests/usecase/Schema02Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema02Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def test_parse(self):
        engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema02.dbs"))
        map = engine.deserialize(
            engine.serialize({"message": {"wasteType": "abcd", "point": {"lat": 47.0, "lng": 15.0}}}))

        self.assertEqual(map, {"message": {"wasteType": "abcd", "point": {"lat": 47.0, "lng": 15.0}, ":type": 0}})


if __name__ == "__main__":
    unittest.main()
